Autopilot/Inc/altimeter.hpp

```diff
@@ -1,69 +1,87 @@
+ * Authors: Lucy Gong, Anthony Berbari
  */
 
 #ifndef ALTIMETER_HPP
 #define ALTIMETER_HPP
 
-#include <ctime>
 #include <cstdint>
 
-struct AltimeterData_t {
+/***********************************************************************************************************************
+ * Definitions
+ **********************************************************************************************************************/
 
-    float pressure, altitude, temp;
+struct AltimeterData_t 
+{
+
+    float altitude;
 
     bool isDataNew; 
-    int status; //TBD but probably 0 = SUCCESS, -1 = FAIL, 1 = BUSY 
-    uint32_t utcTime; //Last time GetResult was called
+    int status;
+};
+
+/**
+ * This union is of no use to the user.
+ * It is for internal use by the module only.
+ * */
+union AltimeterAltitudeRepresentation
+{
+    uint8_t byte[3];
+    int32_t Q16point4;
 };
 
-class Altimeter{
+/***********************************************************************************************************************
+ * Prototypes
+ *********************************************************************************************************************/
+
+class Altimeter {
     public:
         /**
-         * Triggers interrupt for new altimeter measurement - stores raw data in variables and returns right away
+         * Begins a transaction with the Altimeter.
+         * This function is non blocking and returns right away, data will be stored inside the module as it arrives.
+         * To achieve synchronous data, this function must be called synchronously.
          * */
         virtual void Begin_Measuring() = 0; 
 
-        /**GetResult should:
-         * 1. Reset dataIsNew flag
-         * 2. Transfers raw data from variables to struct
-         * 3. Updates utcTime and status values in struct as well
+        /**
+         * Retrieves any data already sent by the altimeter.
+         * If no new data is available, the appropriate flag will be set in the return struct.
+         * All contents of the return struct, apart from the isDataNew flag, are undefined unless isDataNew is set to 1.
+         * This function is non blocking and returns right away.
+         * @param[in]       Data        reference to the results struct.
          * */
-        virtual void GetResult(AltimeterData_t *Data) = 0; //
+        virtual void GetResult(AltimeterData_t &Data) = 0;
 };
 
-class MS5637 : public Altimeter {
+class MPL3115A2 : public Altimeter {
     public:
-        MS5637(const MS5637*) = delete; //Apparently if you try to copy a singleton this will give you errors?
-        static MS5637* GetInstance();
+        /**
+         * This module is built as a singleton. Thus to access an MPL3115A2 object, this function must be called.
+         * Only a single MPL3115A2 object will ever be created and will be shared by all callers of this function.
+         * @return      Altimeter        reference to the singleton object.
+         * */
```

Comment:
I like this comment. 

Note to self: use this as the standard comment for all singletons